pkg/tarutil/tarutil.go

```diff
@@ -78,7 +80,7 @@ func ExtractFiles(r io.Reader, filenames []string) (FilesMap, error) {
 		// Determine if we should extract the element
 		toBeExtracted := false
 		for _, s := range filenames {
-			if strings.HasPrefix(filename, s) {
+			if match, err := regexp.MatchString("^"+s, filename); err == nil && match {
```

Comment:
This leading `^` is implicit behavior and seems unnecessary.
If we have to change other files, I don't mind.

Reply:
The `^` does change the semantics (https://play.golang.org/p/yZl7bJLzr22) - I included it to keep the behaviour in line with the previous behaviour of only checking for the prefix.

I don't think there are any changes needed to the other files as the other file definitions were simple prefixes and didn't contain any special characters, it was more a headsup in case there were other subtle implications.

Reply:
Since we're moving to use the regex, I think it makes sense to enforce all the implementations with `RequiredFile` implementation to use regex.

Reply:
okay, I've removed the `^` and updated all users to have a regexp instead in order to maintain previous behaviour